ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB.hs

```diff
@@ -1,82 +1,1987 @@
+--     read and used to restore the current ledger state, as well as the past
+--     \(k\) ledger states.
+--
+-- - __Flushing 'LedgerTable' differences__: The running Consensus has to
+--     periodically flush chunks of differences from the 'DbChangelog' to the
+--     'BackingStore', so that memory is off-loaded to the backing store, and if
+--     the backing store is an on-disk implementation, reduce the memory usage.
+--
+-- Note that whenever we say /ledger state/ we mean the @'ExtLedgerState' blk
+-- mk@ type described in "Ouroboros.Consensus.Ledger.Basics".
+--
+-- See the implementation description in [Implementation](#g:implementation)
+--
+-- === __(image code)__
+-- >>> import Image.LaTeX.Render
+-- >>> import Control.Monad
+-- >>> import System.Directory
+-- >>>
+-- >>> createDirectoryIfMissing True "docs/haddocks/"
+-- >>> :{
+-- >>> either (error . show) pure =<<
+-- >>>  renderToFile "docs/haddocks/ledgerdb-switch.svg" defaultEnv (tikz ["positioning", "arrows"]) "\
+-- >>> \ \\draw (0, 0) -- (50pt, 0) coordinate (I);\
+-- >>> \  \\draw (I) -- ++(20pt,  20pt) coordinate (C1) -- ++(20pt, 0) coordinate (C2);\
+-- >>> \  \\draw (I) -- ++(20pt, -20pt) coordinate (F1) -- ++(20pt, 0) coordinate (F2) -- ++(20pt, 0) coordinate (F3);\
+-- >>> \  \\node at (I)  {$\\bullet$};\
+-- >>> \  \\node at (C1) {$\\bullet$};\
+-- >>> \  \\node at (C2) {$\\bullet$};\
+-- >>> \  \\node at (F1) {$\\bullet$};\
+-- >>> \  \\node at (F2) {$\\bullet$};\
+-- >>> \  \\node at (F3) {$\\bullet$};\
+-- >>> \  \\node at (I) [above left] {$I$};\
+-- >>> \  \\node at (C1) [above] {$C_1$};\
+-- >>> \  \\node at (C2) [above] {$C_2$};\
+-- >>> \  \\node at (F1) [below] {$F_1$};\
+-- >>> \  \\node at (F2) [below] {$F_2$};\
+-- >>> \  \\node at (F3) [below] {$F_3$};\
+-- >>> \  \\draw (60pt, 50pt) node {$\\overbrace{\\hspace{60pt}}$};\
+-- >>> \  \\draw (60pt, 60pt) node[fill=white] {$k$};\
+-- >>> \  \\draw [dashed] (30pt, -40pt) -- (30pt, 45pt);"
+-- >>> :}
+--
 module Ouroboros.Consensus.Storage.LedgerDB (
-    BackingStore (..)
-  , BackingStorePath (..)
-  , BackingStoreValueHandle (..)
-  , DbChangelog (..)
-  , DbChangelog'
-  , DiskSnapshot (..)
-  , InitFrom (..)
-  , KeySetsReader
-  , LedgerBackingStore
-  , LedgerBackingStore'
-  , LedgerBackingStoreValueHandle
-  , LedgerBackingStoreValueHandle'
-  , LedgerDB (..)
-  , LedgerDBHandle (..)
-  , LedgerDBState (..)
-  , LedgerDBStateEnv (..)
+    -- * The LedgerDB API
+    LedgerDB (..)
+    -- ** Views
+  , LedgerDBView (..)
+  , LedgerDBView'
+  , closeLedgerDBView
+    -- ** Supporting types
+  , SnapCounters (..)
+  , ValidateResult (..)
+    -- ** Constraints for a LedgerDB
   , LedgerDbSerialiseConstraints
-  , PointNotFound (..)
-  , RangeQuery (..)
-  , RewoundTableKeySets (..)
-  , SnapshotFailure (..)
+    -- * Implementation #implementation#
+    --
+    -- | The LedgerDB implementation mainly consists of two components:
```

Comment:
I think it's outside the scope of this PR, but I wonder if we could insert an explanation around this point relating the changelog and backing store, and also expanding on:

- what is the volatile data
- what are differences 
- what are ledger tables 

We don't need to explain these concepts in depth (because they're probably explained somewhere else), just give a rough idea of what they are and illustrate them with some examples.

Reply:
I hyperlinked 'LedgerState', 'Data.Map.Strict.Diff.Diff' and 'LedgerTables'. I think that should be enough, the user can follow the links